src/shapes/Object/Object.ts

```diff
@@ -734,7 +734,7 @@ export class FabricObject<
    */
   hasStroke() {
     return (
-      this.stroke && this.stroke !== 'transparent' && this.strokeWidth !== 0
+      !!this.stroke && this.stroke !== 'transparent' && this.strokeWidth !== 0
```

Comment:
without this it would return empty string
although "falsy" it is not a boolean
qunit was testing this loosely treating empty string as false
but vitest is bit more strict and precise here

Reply:
the intention of this is to be a boolean, can you add it to the typo of hasStroke explicitly?

Reply:
You mean to add :boolean as return type of this method?

Reply:
done